lib/src/Uncertainty/Algorithm/Sensitivity/openturns/RankSobolSensitivityAlgorithm.hxx

```diff
@@ -0,0 +1,132 @@
+  : public SobolIndicesAlgorithmImplementation
+{
+  CLASSNAME
+  friend struct RankSobolBootstrapPolicy;
+public:
+
+  /** Default constructor */
+  RankSobolSensitivityAlgorithm();
+
+  /** Constructor with parameters */
+  RankSobolSensitivityAlgorithm(const Sample & inputDesign,
+                                const Sample & outputDesign);
+
+  /** Virtual constructor */
+  RankSobolSensitivityAlgorithm * clone() const override;
+  
+  /** Method that draw (plot) the sensitivity graph */
+  Graph draw() const override;
+  
+  /** Design accessor */
+  void setDesign(const Sample & inputDesign,
+                 const Sample & outputDesign);
+                                 
+  /** First order indices accessor */
+  Point getFirstOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** String converter */
+  String __repr__() const override;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;  
+  
+  /** Second order indices accessor */
+  SymmetricMatrix getSecondOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+  
+  /** Total order indices accessor */
+  Point getTotalOrderIndices(const UnsignedInteger marginalIndex = 0) const override;
+
+  /** Interval for the total order indices accessor */
+  Interval getTotalOrderIndicesInterval() const override;
+  
+  /** Aggregated total order indices accessor for multivariate samples */
+  Point getAggregatedTotalOrderIndices() const override;
+  
+  /** Whether to use bootstrap or asymptotic distribution */
+  void setUseAsymptoticDistribution(Bool ) override {};
+  Bool getUseAsymptoticDistribution() const override;
+
+  /** Estimator distribution accessor */
+  Distribution getTotalOrderIndicesDistribution() const override;
+   
+protected:
+
+  Sample computeIndices() const;
+  
+  /** void method that computes bootstrap confidence interval */
+  void computeBootstrapDistribution() const override;
+
+private:
+
+  static Graph DrawSobolFirstIndices(const Description & inputDescription,
+                                     const Point & firstOrderIndices,
+                                     const Interval & firstOrderConfidenceInterval = Interval());
+
+  /** Internal method that compute Vi/VTi using a sample */
+  Sample computeIndices(const Sample & sample, Sample & VTi) const override;
+  
+  Sample computeIndicesFromSample(const Sample &, const Sample &) const;
+  
+  /** Function that computes aggregated indices using Vi/VTi + variance  */
+  Point computeAggregatedIndices(const Sample & Vi,
+                                 const Point & variance) const;
+                                                        
+                                
+  void setDesign(const Sample & ,
+                 const Sample & ,
+                 const UnsignedInteger ) override {};
```

Comment:
no need to ignore this one just because you dont need the last argument ;
you could reuse it and just ignore the last argument instead of reimplementing it with just 2 arguments

Reply:
Actually, this is not the same setDesign method. The method from the parent class involves the test "
`  if (outputDesign.getSize() < size * (inputDimension + 2))
    throw InvalidArgumentException(HERE) << "Sobol experiment is too small (" << outputDesign.getSize()
                                         << " vs " << size * (inputDimension + 2) << ")";` which is not relevant here and will cause failure with this data-driven algorithm.

Reply:
yes, but you can re-implement it in RankSobol with the same signature instead of declaring 2 different setDesign method

Reply:
okay.